test/unit/tuple_test/device/tuple.hpp

```diff
@@ -0,0 +1,396 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright 2021-2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef ROCWMMA_DEVICE_TUPLE_TEST_HPP
+#define ROCWMMA_DEVICE_TUPLE_TEST_HPP
+
+#include <rocwmma/internal/io_traits.hpp>
+#include <rocwmma/internal/layout.hpp>
+#include <rocwmma/internal/tuple.hpp>
+#include <rocwmma/rocwmma.hpp>
+
+static constexpr uint32_t ERROR_VALUE   = 7u;
+static constexpr uint32_t SUCCESS_VALUE = 0u;
+
+namespace rocwmma
+{
+    template <typename DataT, uint32_t VecSize>
+    __device__ static inline DataT get(VecT<DataT, VecSize> const& v, uint32_t idx)
+    {
+        return v.data[idx];
+    }
+
+    template <typename DataT, uint32_t VecSize>
+    __device__ static inline bool isTupleTest()
+    {
+        bool err = false;
+
+        err |= std::is_tuple<int>::value;
+        err |= !std::is_tuple<std::tuple<int, float>>::value;
+        // err |= !std::is_tuple<const std::tuple<int, float>>::value;
```

Comment:
`const tuple` and `tuple &` cannot pass the test

Reply:
Probably will need to add some definitions there

Reply:
> Probably will need to add some definitions there

It makes sense if the tuple is only used in meta-programming. Const tuple or tuple ref should not be used. 

So I can change the check to `err |= std::is_tuple<const std::tuple<int, float>>::value;`

What do you think?

Reply:
Ok, I agree